src/models/kafkaCluster.ts

```diff
@@ -3,33 +3,39 @@ import * as vscode from "vscode";
 import { CCLOUD_CONNECTION_ID, IconNames, LOCAL_CONNECTION_ID } from "../constants";
 import { CustomMarkdownString } from "./main";
 
-/** Main class representing a local Kafka cluster */
-export class LocalKafkaCluster extends Data {
-  readonly connectionId: string = LOCAL_CONNECTION_ID;
-  readonly isLocal: boolean = true;
-  readonly isCCloud: boolean = false;
+/** Base class for all KafkaClusters, be they local or be they CCloud */
+export abstract class KafkaCluster extends Data {
+  abstract readonly connectionId: string;
+  abstract readonly isLocal: boolean;
+  abstract readonly isCCloud: boolean;
+
+  abstract name: string;
 
   id!: Enforced<string>;
   bootstrapServers!: Enforced<string>;
   uri!: Enforced<string>;
+}
+
+/** A local Kafka cluster */
```

Comment:
We could probably remove this and the `CCloudKafkaCluster` comment since they don't add much context